### PR TITLE
Make IDE usage entirely optional #6973

### LINKED ISSUE #6787: Intellij setup: git error reported at start up
This error is shown at start up
image

More details:
image

### LINKED ISSUE #6818: Add instructions for restoring IntelliJ run configurations
**Problem**: IntelliJ run configurations are removed by IntelliJ upon importing the project.

**Solution**: Add instructions to restore run configurations using git after importing the project.

### LINKED ISSUE #6973: Make IDE usage entirely optional
Helpful as IDEs may, their usage should never be a requirement because we have a working workflow without the need of one, and any CI does not assume the use of one either.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #6973

This is the PR that will finally release TEAMMATES' status as an "Eclipse-only project". All important operations such as building, testing, and deploying are all supported without any IDE, with some enhancements introduced via this PR.

**Outline of Solution**

- Updated documentation to clearly show that IDE is optional albeit recommended.
- Added a dedicated documentation for IDE setup; Eclipse and IntelliJ, and made them more detailed step-by-step.
- Added Gradle commands to:
  - Provide test configuration files for Eclipse and IntelliJ. Thus, this will fix #6818 because the run configs are no longer in source control.
  - Run all other test types via command line: local tests, failed tests, any individual test
- Removed "Staging tests"; what is it used for?

This also fixes #6787 because all tracked `.gitignore`-d files have been fixed either by fixing the entry in `.gitignore` or removing the files to properly source-controlled folders.

> In case anyone thinks I'm anti-IDE: of course I will stay using Eclipse to do stuffs. It makes many things tremendously easier; why would I trouble myself? The problem lies in _making IDEs compulsory_, not _using IDE_.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
I'll need some review helps here:
@samsontmr - to make sure that the IntelliJ instructions are still legitimate. Also, how do you run single tests on IntelliJ?
@whipermr5 - as a general devops apprentice :joy:

File: docs/ide-usage.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+     Configure for all the languages used in TEAMMATES:
+     * Java: `Java → Code Style → Formatter → Edit → Tab policy → Spaces only`.
+     * JavaScript: `JavaScript → Code Style → Formatter → Edit → Tab policy → Spaces only`.
+     * HTML: `Web → HTML Files → Editor → Indent using spaces`.
+     * CSS: `Web → CSS Files → Editor → Indent using spaces`.
+     * XML: `XML → XML Files → Editor → Indent using spaces`.
+   * Validation: We do not validate HTML, JSP, and XML.<br>
+     `Validation` → uncheck the `Build` option for `HTML Syntax Validator`, `JSP Content Validator`, `JSP Syntax Validator`, and `XML Validator`.
+
+1. `Clean` the project for all changes to take effect. Ensure that there is no errors. Warnings are generally fine to be ignored.
```
Comment (whipermr5) [inline docs/ide-usage.md]: `there is no` -> `there are no`
`Warnings are generally fine to be ignored` -> `Warnings are generally fine and can be ignored`

File: docs/development.md [hunk 2]
```diff
@@ -185,41 +188,47 @@ Before running the test suite, both the server and the test environment should b
...
+To run individual tests, right-click on the test files on the project explorer and choose `Run As → TestNG Test`.
 
-1. Run the following command to run the full test suite once and retry the failed tests several times:
-   ```sh
-   ./gradlew ciTests
-   ```
+#### IntelliJ
 
-1. The final result can be viewed by opening `{project folder}/build/reports/test-try-{n}/index.html`, where `{n}` is the sequence number of the test run.
+Run tests using the configurations available under `Run → Play`.
```
Comment (samsontmr) [inline docs/development.md]: `Play` -> `Run...`

File: docs/development.md [hunk 3]
```diff
@@ -185,41 +188,49 @@ Before running the test suite, both the server and the test environment should b
...
+To run individual tests, right-click on the test files on the project explorer and choose `Run As → TestNG Test`.
 
-1. Start the dev server.
+#### IntelliJ
 
-1. Run the following command to run the full test suite once and retry the failed tests several times:
-   ```sh
-   ./gradlew ciTests
-   ```
+Run tests using the configurations available under `Play → Run`.
```
Comment (samsontmr) [inline docs/development.md]: Oops looks like my correction was not totally clear, it should be exactly `Run → Run...` with the ellipses.

**Pull request merged**